doc/reference/caches.rst

```diff
@@ -1,16 +1,41 @@
-**********
-Dune Cache
-**********
+Dune Caches
+===========
+
+Dune implements several different caches:
+
+- The build cache: contains build artifacts that are shared across different
+  workspaces. This is the main cache, interacted with with the ``dune cache``
+  command.
+- The toolchains cache: contains the compiler (and compiler variants) when they
+  can't be installed as standard packages.
+- The revision store: contains packaging information fetched from package
+  repositories (`e.g.` ``opam-repository``).
+- The LMDB cache: this is a subcache of the revision store, caching expensive
+  git lookup operations.
 
 .. TODO(diataxis) This is reference material with some explanation.
 
-Dune implements a cache of build results that is shared across different
-workspaces. Before executing a build rule, Dune looks it up in the shared
+Before executing a build rule, Dune looks it up in the shared
 cache, and if it finds a matching entry, Dune skips the rule's execution and
```

Comment:
I think the issues @Leonidas-from-XIV points out here is a result of 'it' being used with two different referents: first to 'a build rule' then to `dune` (or dune's cache looking up).